newsrevenuehub/rev-engine
# #5577 Audit Trail

#### What's this PR do?
This pull request creates an audit trail of changes to various models across the project:
 - `django-simple-history` has been added to the project to track changes for each model
 - changes are visible on each object's detail page in the Django admin (for example, at the /nrhadmin/organizations/organization/1/history/ URL)
 - a new `RevEngineHistoricalChange` model keeps track of all revisions, and exposes them in a Django admin view (at the /nrhadmin/common/revenginehistoricalchange/ URL), showing admin users:
   - who made each change
   - when the change occurred
   - what type of change occurred (create, update, delete)
   - which fields changed during an update
 - a Django signal exists to create a `RevEngineHistoricalChange` object when `django-simple-history` tries to create a historical record

#### How should this be manually tested?
 - go to a model’s detail page in the Django admin (for example, at the /nrhadmin/organizations/organization/1/change/ URL)
 - make a change (for example, edit the ‘Salesforce ID’), save the object, and then click the “HISTORY” button in the top right corner of the object’s detail page in the Django admin
 - observe a list of all of the changes to the model, including the change made, the datetime for each change, the type of change made, and the user who made the change
 - go to the list of all tracked changes in the Django admin (at the /nrhadmin/common/revenginehistoricalchange/ URL)
 - observe the change you just made in the list view
 - make any other changes (in the Django admin or in the app, such as to a donation page), and observe them in the list view at the /nrhadmin/common/revenginehistoricalchange/ URL

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?
No
